tests/test_classes.py

```diff
@@ -213,30 +204,30 @@ def test_stringify_func_is_importable(self):
 
         assert set_flag == func
 
-    def test_can_get_job_details(self, inmem_client, scheduled_job):
-        assert inmem_client.status(
-            scheduled_job.job_id).job_id == scheduled_job.job_id
+    def test_can_get_job_details(self, inmem_queue, enqueued_job):
+        assert inmem_queue.fetch_job(
+            enqueued_job.job_id).job_id == enqueued_job.job_id
 
-    def test_can_cancel_a_job(self, inmem_client):
-        is_running_event = EventProxy()
-        is_not_canceled_event = EventProxy()
-        job_id = inmem_client.schedule(
+    def test_can_cancel_a_job(self, inmem_queue):
+        job_id = inmem_queue.enqueue(
             cancelable_job,
-            is_running_event=is_running_event,
-            is_not_canceled_event=is_not_canceled_event,
             cancellable=True
         )
 
-        is_running_event.wait(1.0)
+        job = inmem_queue.fetch_job(job_id)
+        while job.state != State.RUNNING:
```

Comment:
We should add a timeout here, or a catch so we don't get into an infinite loop.

Reply:
Fair enough.